leo_gz_plugins/src/differential_system.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Fictionlab sp. z o.o.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#ifdef USE_IGNITION
+#define ADD_PLUGIN IGNITION_ADD_PLUGIN
+#include <ignition/gazebo/System.hh>
+#include <ignition/gazebo/Model.hh>
+#include <ignition/gazebo/components.hh>
+#include <ignition/plugin/Register.hh>
+#define  gazebo ignition::gazebo
+#define error ignerr
+#define msg ignmsg
+#else
+#define ADD_PLUGIN GZ_ADD_PLUGIN
+#include <gz/sim/System.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/components.hh>
+#include <gz/plugin/Register.hh>
+#define  gazebo gz::sim
+#define error gzerr
+#define msg gzmsg
+#endif
```

Comment:
Gazebo garden provides backward compatibility with the ignition headers and macros. I think we should make it simpler by using the compatibility as long as we support Gazebo Fortress.

My proposal:
```suggestion
#define SUPPRESS_IGNITION_HEADER_DEPRECATION

#include <ignition/gazebo/System.hh>
#include <ignition/gazebo/Model.hh>
#include <ignition/gazebo/components.hh>
#include <ignition/plugin/Register.hh>

#ifdef USE_IGNITION
namespace gazebo = ignition::gazebo;
#else
namespace gazebo = gz::sim;
#endif

```